packages/babel-parser/src/parser/expression.js

```diff
@@ -1138,6 +1139,25 @@ export default class ExpressionParser extends LValParser {
           this.registerTopicReference();
           return this.finishNode(node, "PipelinePrimaryTopicReference");
         }
+
+        if (isIdentifierStart(this.input.codePointAt(this.state.end))) {
```

Comment:
This is insufficient as the first character of an identifier could be a unicode escape sequence.
```js
class Foo {
  #a;
  static isFoo(obj) {
    return #\u{61} in obj;
  }
}
```